h2/src/main/org/h2/store/fs/encrypt/FileEncrypt.java

```diff
@@ -73,9 +73,19 @@ public FileEncrypt(String name, byte[] encryptionKey, FileChannel base) {
         this.encryptionKey = encryptionKey;
     }
 
-    private void init() throws IOException {
+    private XTS init() throws IOException {
```

Comment:
why XTS instance has to be passed on as result / local variable? existing field will do just fine

Reply:
never mind, i got it, it saves one volatile read

Reply:
Actually IMHO, @little-pan change (final field) is more appropriate. How often this object is created? How likely, that it won't be used afterwards, so lazy initialization pays off?

Reply:
In some virtual environments (but not on Travis CI) file locks don't work as expected and alternative locking mechanisms are really necessary (I have such VPS provided by web hosting company). Initialization of `XTS` performs some I/O, but file can be not yet really locked, therefore `FileEncrypt` **intentionally** delays its initialization and this is documented in its constructor. I'm not the author of this code and I have no idea whether this logic is still required by PageStore and MVStore backends and, unfortunately, I can't test it now. But I'm sure that we need to perform real tests in these environments first, otherwise we can create a regression in sensitive area of H2.

Reply:
Actually I don't want to test anything on production servers for oblivious reasons and I don't have testing systems with this problem.

These VPS work in OpenVZ containers, but I don't know is it a problem of OpenVZ or a problem of underlying file system (possibly distributed one, I don't have any information about it).

Reply:
sometimes it pays off to read comments :smile: